### PR TITLE
Feature/wa 271 todo fixes

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Fixed some todos

### CODE REVIEW

File: safeUIKit/safeUIKit/TextInput.swift [hunk 1]
```diff
@@ -96,6 +100,10 @@ public final class TextInput: UIView {
         layer.add(CABasicAnimation.shake(center: center), forKey: TextInput.shakeAnimationKey)
     }
 
+    @objc private func textChanged(_ sender: Any) {
+        text = textField.text // validation
```
Comment (sche) [inline safeUIKit/safeUIKit/TextInput.swift:104]: Will text field input stay visually with 100 chars in this case?

File: safeUIKit/safeUIKit/TextInputTests.swift [hunk 2]
```diff
@@ -108,6 +108,19 @@ class TextInputTests: XCTestCase {
...
+        input.maxLength = 1
+        input.text = "11"
+        XCTAssertEqual(input.text, "1")
+    }
+
+    func test_whenTryingToTypeMoreThanLength_thenTakesFirstNChars() {
+        input.maxLength = 1
+        input.textField.text = "abc"
+        input.textField.sendActions(for: .editingChanged)
+        XCTAssertEqual(input.text, "a")
```
Comment (sche) [inline safeUIKit/safeUIKit/TextInputTests.swift:121]: textField should also have "a"

File: Common/Common/BaseID.swift [hunk 3]
```diff
@@ -11,7 +11,7 @@ open class BaseID: Hashable, Assertable {
     }
 
     public let id: String
-    public var hashValue: Int { return id.hashValue * 31 }
+    public var hashValue: Int { return id.hashValue &* 31 } // hashValue * 31 may overflow, so using &* operator
```
Comment (sche) [inline Common/Common/BaseID.swift:14]: wow :)

File: safe/safe/FatalErrorHandler.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        FatalErrorHandler().showFatalError(message: message, log: log, error: error)
+    }
+
+    private func showFatalError(message: String, log: String, error: Error?) {
+        ApplicationServiceRegistry.logger.fatal(log, error: error)
+        let window = UIWindow(frame: UIScreen.main.bounds)
+        let vc = UIViewController()
+        vc.view.backgroundColor = .clear
+        window.rootViewController = vc
+        window.windowLevel = UIWindowLevelAlert + 1
```
Comment (sche) [inline safe/safe/FatalErrorHandler.swift:29]: wow :)

**Review by sche — APPROVED** [review]

File: safeUIKit/safeUIKit/TextInputTests.swift (hunk 2 above)
Comment (DmitryBespalov) [reply safeUIKit/safeUIKit/TextInputTests.swift:121]: the text is a proxy for textfield.text

File: safeUIKit/safeUIKit/TextInput.swift (hunk 1 above)
Comment (DmitryBespalov) [reply safeUIKit/safeUIKit/TextInput.swift:104]: yes, textfield is not allowing input of more than 100 chars, it cuts everything above that limit

**Pull request merged**